### DISCUSSION TITLE
Loading a saved casadi function object in matlab

### DISCUSSION THREAD

**Discussion by vasu3code** [casadi/casadi#3385] [Q&A]
Hi,

I am trying to import a saved function object created in python in MATLAB. My aim is to create a neural network to imitate a process model and then approximate the same as a casadi function and later create a MPC formulation in MATLAB using the imported casadi object.

This work is inspired by ml-casadi (https://github.com/TUM-AAS/ml-casadi.git). 

I created a simple network (just a simple MLP ) and then saved the corresponding function as a casadi function object using casadi_func.save.

I now want to import this in matlab and use it as a plant model later using symbolic notations for setting up an MPC formulation using acados. Could someone guide me how I could import this function object directly in MATLAB?
```

## Create a randomly initialized non linear Multi Layer Perceptron
    model2 = mc.nn.MultiLayerPerceptron(2, 3, 1, 1, 'Tanh')

    ## Export the model as Casadi Function
    casadi_sym_inp = cs.MX.sym('inp', 2)
    casadi_sym_out = model2(casadi_sym_inp)
    casadi_func = cs.Function('model2',
                              [casadi_sym_inp],
                              [casadi_sym_out])
    casadi_func.save('casadi_func')
```
TIA!

**Comment by jgillis** [comment]
Did you try casadi.Function.load(...)?

**Reply by vasu3code** [reply]
Hi,

It seemed to not work. I have a really simple script right now.

```
import casadi.*
 
f = casadi.function.load('/home/vasu/Documents/work/ml-casadi/examples/casadi_func');
```

The repository contains a casadi_func file and running in matlab gave the following error :(

```
Unable to resolve the name casadi.function.load.

Error in casadi_import (line 3)
f = casadi.function.load('/home/vasu/Documents/work/ml-casadi/examples/casadi_func');
```

**Reply by jgillis** [reply]
Note the capital 'F'

**Reply by vasu3code** [reply]
Changed the same but still got the same result. 

```
Unable to resolve the name casadi.Function.load.

Error in casadi_import (line 3)
F = casadi.Function.load('/home/vasu/Documents/work/ml-casadi/examples/casadi_func');
```

**Reply by jgillis** [reply]
Did you get CasADi to work at all in Matlab?
